workspaces/api-gateway/src/api/v1/payments/services/CreatePaymentService.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import type { Repository } from 'typeorm';
+
+import { Payment } from 'database/models/payment.entity';
+import { CreatePaymentDetailService } from 'api/v1/payments/services/CreatePaymentDetailService';
+import type { ICreatePaymentParams } from 'api/v1/payments/interface';
+
+@Injectable()
+export class CreatePaymentService {
+  constructor(
+    @InjectRepository(Payment)
+    private paymentRepository: Repository<Payment>,
+    private createPaymentDetailService: CreatePaymentDetailService,
+  ) {}
+
+  public async exec({
+    userId,
+    methodId,
+    listCartDetails,
+    amount,
+    metadata,
+  }: ICreatePaymentParams): Promise<Payment> {
+    const newPayment = this.paymentRepository.create({
+      userId,
+      metadata,
+      methodId,
+      amount,
+    });
+
+    try {
+      await this.paymentRepository.save(newPayment);
+
+      listCartDetails.map(async (addProduct) => {
+        const { productId, quantity, product } = addProduct;
+        await this.createPaymentDetailService.exec({
+          paymentId: newPayment.id,
+          productId,
+          quantity,
+          unitPrice: product.price,
+        });
+      });
```

Comment:
why we use the map here?

Reply:
để tạo list payment detail từ list product người dùng chọn để checkout ạ

Reply:
and what is the map for?